presto-hive/src/main/java/com/facebook/presto/hive/rule/HivePlanOptimizerProvider.java

```diff
@@ -13,24 +13,53 @@
  */
 package com.facebook.presto.hive.rule;
 
+import com.facebook.presto.hive.HivePartitionManager;
+import com.facebook.presto.hive.HiveTransactionManager;
 import com.facebook.presto.spi.ConnectorPlanOptimizer;
 import com.facebook.presto.spi.connector.ConnectorPlanOptimizerProvider;
+import com.facebook.presto.spi.function.FunctionMetadataManager;
+import com.facebook.presto.spi.function.StandardFunctionResolution;
+import com.facebook.presto.spi.relation.RowExpressionService;
 import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
 
 import java.util.Set;
 
+import static java.util.Objects.requireNonNull;
+
 public class HivePlanOptimizerProvider
         implements ConnectorPlanOptimizerProvider
 {
+    private final HiveTransactionManager transactionManager;
+    private final RowExpressionService rowExpressionService;
+    private final StandardFunctionResolution functionResolution;
+    private final HivePartitionManager partitionManager;
+    private final FunctionMetadataManager functionMetadataManager;
+
+    @Inject
+    public HivePlanOptimizerProvider(
+            HiveTransactionManager transactionManager,
+            RowExpressionService rowExpressionService,
+            StandardFunctionResolution functionResolution,
+            HivePartitionManager partitionManager,
+            FunctionMetadataManager functionMetadataManager)
+    {
+        this.transactionManager = requireNonNull(transactionManager, "transactionManager is null");
+        this.rowExpressionService = requireNonNull(rowExpressionService, "rowExpressionService is null");
+        this.functionResolution = requireNonNull(functionResolution, "functionResolution is null");
+        this.partitionManager = requireNonNull(partitionManager, "partitionManager is null");
+        this.functionMetadataManager = requireNonNull(functionMetadataManager, "functionMetadataManager is null");
+    }
+
     @Override
     public Set<ConnectorPlanOptimizer> getLogicalPlanOptimizers()
     {
-        return ImmutableSet.of();
+        return ImmutableSet.of(new HiveFilterPushdown(transactionManager, rowExpressionService, functionResolution, partitionManager, functionMetadataManager));
     }
 
     @Override
     public Set<ConnectorPlanOptimizer> getPhysicalPlanOptimizers()
     {
-        return ImmutableSet.of();
+        return ImmutableSet.of(new HiveFilterPushdown(transactionManager, rowExpressionService, functionResolution, partitionManager, functionMetadataManager));
```

Comment:
Add a comment to explain "new filters can be created and we need to merge them together" something like that